widgets/pie/pie_test.go

```diff
@@ -0,0 +1,135 @@
+package pie
+
+import (
+	"image"
+	"testing"
+
+	"github.com/mum4k/termdash/cell"
+	"github.com/mum4k/termdash/private/canvas/braille"
+	"github.com/mum4k/termdash/private/canvas/testcanvas"
+	"github.com/mum4k/termdash/private/faketerm"
+	"github.com/mum4k/termdash/widgetapi"
+)
+
+func TestPie(t *testing.T) {
+	tests := []struct {
+		desc          string
+		values        []int
+		colors        []cell.Color
+		canvas        image.Rectangle
+		wantDrawErr   bool
+		wantNewErr    bool
+		wantValuesErr bool
+	}{
+		{
+			desc:       "New fails with no options",
+			canvas:     image.Rect(0, 0, 5, 5),
+			wantNewErr: false,
+		},
+		{
+			desc:          "Values fails with empty values",
+			values:        []int{},
+			colors:        []cell.Color{cell.ColorRed},
+			canvas:        image.Rect(0, 0, 5, 5),
+			wantValuesErr: true,
+		},
+		{
+			desc:          "Values fails with negative value",
+			values:        []int{10, -5},
+			colors:        []cell.Color{cell.ColorRed, cell.ColorBlue},
+			canvas:        image.Rect(0, 0, 5, 5),
+			wantValuesErr: true,
+		},
+		{
+			desc:   "Draws pie chart with valid values and colors",
+			values: []int{10, 20},
+			colors: []cell.Color{cell.ColorRed, cell.ColorBlue},
+			canvas: image.Rect(0, 0, 10, 10),
+		},
+		{
+			desc:        "Fails to draw when canvas is too small",
+			values:      []int{10, 20},
+			colors:      []cell.Color{cell.ColorRed, cell.ColorBlue},
+			canvas:      image.Rect(0, 0, 1, 1),
+			wantDrawErr: true,
```

Comment:
(optional) Consider adding a few test cases that draw on the unit test only canvas.

Example from the donut widget:

https://github.com/mum4k/termdash/blob/102df20a882547528c83047688595acd2a82c7ef/widgets/donut/donut_test.go#L271-L286

This would likely require using the draw.BrailleCircle function suggested above.